.github/workflows/ci.yml

```
@@ -103,6 +103,8 @@ jobs:
 
   slither-analyze:
     runs-on: ubuntu-latest
+    permissions:
+      security-events: write
```

Do we need `contents: read` also? That's what they show in the slither-action readme: https://github.com/crytic/slither-action#example-workflow-hardhat-and-sarif

```suggestion
    permissions:
      contents: read
      security-events: write
```

---

I don't think it's necessary. I did have `contents: read` set in my repo last week, but CI still runs successfully after I removed it. I'm inclined to think that it's redundant.

---

Got it, I'm ok with leaving it out for now and can always change if it's an issue. Thanks!